src/dbms/inmemory/replication_handlers.cpp

```diff
@@ -329,6 +334,78 @@ void InMemoryReplicationHandlers::SnapshotHandler(dbms::DbmsHandler *dbms_handle
   spdlog::debug("Replication recovery from snapshot finished!");
 }
 
+void InMemoryReplicationHandlers::ForceResetStorageHandler(dbms::DbmsHandler *dbms_handler,
+                                                           const std::optional<utils::UUID> &current_main_uuid,
+                                                           slk::Reader *req_reader, slk::Builder *res_builder) {
+  storage::replication::ForceResetStorageReq req;
+  slk::Load(&req, req_reader);
+  auto db_acc = GetDatabaseAccessor(dbms_handler, req.db_uuid);
+  if (!db_acc) {
+    storage::replication::SnapshotRes res{false, 0};
+    slk::Save(res, res_builder);
+    return;
+  }
+  if (!current_main_uuid.has_value() || req.main_uuid != current_main_uuid) [[unlikely]] {
+    LogWrongMain(current_main_uuid, req.main_uuid, storage::replication::SnapshotReq::kType.name);
+    storage::replication::SnapshotRes res{false, 0};
+    slk::Save(res, res_builder);
+    return;
+  }
+
+  storage::replication::Decoder decoder(req_reader);
+
+  auto *storage = static_cast<storage::InMemoryStorage *>(db_acc->get()->storage());
+
+  auto storage_guard = std::unique_lock{storage->main_lock_};
```

Comment:
I would rather use `db.try_exclusively([](auto &in) {...});`
That way we know we are absolutely the only one handling this database.

Reply:
I think there is no need as we are only transaction due to stream and we have `main_lock_`

Reply:
Transactions and database access are different levels of concern.
I think storage/transaction level is correct here.
But its problematic that an RPC call can be held up forever by any other current transactions.

Reply:
Yes, you are right, makes sense

Reply:
I think it is a general issue we have, with snapshots also, so we need to deal with both cases in different way. I would leave it as TODO in case it is the real issue